test/test_lib.py

```diff
@@ -578,12 +588,12 @@ def test_others_found(self):
             ops.lib.use('baz', 2, 'bob@example.com')
 
 
-class TestDeprecationWarning(TestCase):
+class TestDeprecationWarning:
```

Comment:
Is there any reason this is still a class? Or is moving it to a couple of top-level tests something for a future phase?

Reply:
Because this test file contained multiple classes, I think mainly for organizing test cases into groups. I kept them untouched. See this comment here: https://github.com/canonical/operator/pull/1195#discussion_r1574116911.